client/dashboard/utils/site-slug.ts

```diff
@@ -0,0 +1,10 @@
+import type { Site } from '../data/types';
+
+export function getSiteSlug( site: Site ) {
+	let url = site.URL;
+	if ( site.options?.is_redirect && site.options?.unmapped_url ) {
+		url = site.options?.unmapped_url;
```

Comment:
Out of curiosity: what is site.URL in the case of a redirect, and why can't we use it as the site slug? Will the REST API not recognise it?